server/container_create_linux.go

```diff
@@ -245,6 +245,12 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 
 	imageName := imgResult.SomeNameOfThisImage
 	imageID := imgResult.ID
+
+	// Avoid image removal during container creation
+	imageRemovalLock := s.StorageImageServer().ImageRemovalLock(imageID)
+	imageRemovalLock.RLock()
+	defer imageRemovalLock.RUnlock()
```

Comment:
can we move this below the container storage call? after we mount the image, the GC will fail, so we can continue with image deltion attempts and report an error to the kubelet (instead of stalling and potentially timing out the gc request)

Reply:
this being the unlock

Reply:
Yes, like this now?